x-pack/plugins/security_solution/public/common/components/ml/anomaly/use_anomalies_search.ts

```diff
@@ -129,21 +131,40 @@ function formatResultData(
     key: string;
     doc_count: number;
   }>,
-  notableAnomaliesJobs: SecurityJob[]
+  notableAnomaliesJobs: SecurityJob[],
+  spaceId: string | undefined
 ): AnomaliesCount[] {
-  const unsortedAnomalies: AnomaliesCount[] = NOTABLE_ANOMALIES_IDS.map((notableJobId) => {
-    const job = notableAnomaliesJobs.find(({ id }) => id === notableJobId);
-
-    const bucket = buckets.find(({ key }) => key === job?.id);
-    const hasUserName = has("entity.hits.hits[0]._source['user.name']", bucket);
-
-    return {
-      name: job?.customSettings?.security_app_display_name ?? notableJobId,
-      count: bucket?.doc_count ?? 0,
-      entity: hasUserName ? AnomalyEntity.User : AnomalyEntity.Host,
-      job,
-    };
+  const unsortedAnomalies: AnomaliesCount[] = NOTABLE_ANOMALIES_IDS.flatMap((notableJobId) => {
+    const jobs = notableAnomaliesJobs.filter(({ id }) => matchJobId(id, notableJobId, spaceId));
+
+    return jobs.map((job) => {
+      const bucket = buckets.find(({ key }) => key === job?.id);
+      const hasUserName = has("entity.hits.hits[0]._source['user.name']", bucket);
+
+      return {
+        name: job.customSettings?.security_app_display_name ?? notableJobId,
+        count: bucket?.doc_count ?? 0,
+        entity: hasUserName ? AnomalyEntity.User : AnomalyEntity.Host,
+        job,
+      };
+    });
   });
-
   return sortBy(['name'], unsortedAnomalies);
 }
+
+export const installedJobPrefix = (spaceId: string | undefined) => `${spaceId ?? 'default'}_`;
+
+export const uninstalledJobIdToInstalledJobId = (
+  moduleJobId: string,
+  spaceId: string | undefined
+) => `${installedJobPrefix(spaceId)}${moduleJobId}`;
+
+/**
+ * From version 8.8, jobs installed using security solution have the spaceId in their name.
+ * Jobs installed using security solution on versions older than 8.8 don't have the spaceId in their name.
+ */
+const matchJobId = (
+  jobId: string,
+  notableJobId: NotableAnomaliesJobId,
+  spaceId: string | undefined
+) => jobId === uninstalledJobIdToInstalledJobId(notableJobId, spaceId) || jobId === notableJobId;
```

Comment:
Would it make sense to pass entire the job object so we can check if the job is installed to do the raw (not installed) comparison or the space-aware (installed) comparison? 
This way we prevent collisions between job and space ids (let's imagine we have a job starting with `default` or a space called `packetbeat`)

Reply:
I had forgotten this. But the corner cause was implemented on purpose. Users can install a module on version 7.7 and later on install the same module on version 7.8. So we would have two versions of the same job: ' default_packetbeat_dns_tunneling` and `packetbeat_dns_tunneling`. I opted to display both in the UI instead of hiding one.

Reply:
I see, that makes sense 👍 
However, false positives can still happen if we have slightly different job names, only changing the first word. But this is out of the scope of this function. I think it would have been better to choose a different separator for the space prefix, the underscore `_` is used to separate words in the job name. But anyway, that's another topic.